docs/CrossCompilationModel.md

```diff
@@ -0,0 +1,87 @@
+Notes, module maps, wrapper types, or Swift `extension`s. This code may or may
+not be fully inlined into the client code and thus be part of the platform ABI.
+
+## Flags
+
+The compiler resources are controlled via the driver flag `-resource-dir`.
+This allows the driver to select the correct location in most cases while
+allowing the developer control to override the value if required.
+Normally, you should not need to set this flag as the location of these files is intrinsic to the compiler.
+
+The system headers are more interesting. Since this is C/C++ content, this is
+actually consumed through the clang importer rather than the Swift compiler. The
+Swift toolchain uses clang as the C/C++ compiler on all platforms as it is
+embedded to generate inline FFI to enable seamless C/C++ bridging to Swift. The
+flag used by clang is derived from the GCC toolchain, and is spelt `--sysroot`.
+The compiler driver is responsible for identifying the structure of the sysroot. When cross-compiling, there isn't a consistent location for these files, so the driver must expose an argument to specify where to find these files.
+sysroot, although because there is no uniform installation location, this
+parameter needs to be exposed to the user to allow selection, particularly in
+the case of cross-compilation.
+
+Currently, we do not have a good way to isolate the SDK overlay from the
+remainder of the required content. On Darwin platforms, this content is shipped
+as part of the SDK. As a result the singular `-sdk` flag allows control over the
+platform SDK and SDK overlay. Windows uses a split model as the Windows SDK is
+split into multiple components and can be controlled individually (i.e. UCRT
+version, SDK version, VCRuntime version). The `-sdk` flag is used to specify the
+location of the SDK overlay which is applied to the system SDK. By default, the
+environment variable `SDKROOT` is used to seed the value of `-sdk`, though the
+user may specify the value explicitly. Other platforms do not currently have a
+flag to control this location and the toolchain defaults to a set of relative
+paths to locate the content. This prevents cross-compilation as the included
+content would be for a single platform.
+
+> [!NOTE]
+> `-resource-dir` historically influenced the driver's search for the SDK content, locating the standard library relative to the resource directory. This behaviour is considered deprecated but remains for compatibility. The `-sdk` parameter is given precedence and is the preferred mechanism for controlling the behaviour of the driver to locate the SDK content.
+
+## Solution
+
+Generalising the above structure and sharing the common sharing gives way to the
+following set of flags for cross-compilation:
+
+1. `-target`: specifies the triple for the host
+2. `-sysroot`: specifies the (C/C++) sysroot for the host platform content
+3. `-sdk`: specifies the Swift SDK overlay for the host
```

Comment:
Do we want to call it an SDK overlay if it is the Swift SDK for the platform?
It contains the C/C++ overlay for importing into Swift, sure, but it usually also contains Foundation, the standard library, etc... (arguably, Foundation could be its own library in its own directory structure and let the standard library exist in its own space, but that's an argument for a different venue.)

Reply:
I mean it still does overlay the platform SDK. I don't like Swift SDK because it can be confusing when you are talking about Darwin vs non-Darwin as the former will use a single unified SDK.